### PR TITLE
feat: support for GEODIST added

### LINKED ISSUE #1547: Add GEODIST support
See here: https://redis.io/commands/geodist/

### PR DESCRIPTION
**Pull request by yoelsherwin**
resolves #1547

### CODE REVIEW

**Comment by yoelsherwin** [comment]
Finished this on,e wasn't sure where should I put the `ExtractUnitOrReply` function

File: src/server/zset_family.cc [hunk 1]
```diff
@@ -2393,6 +2393,21 @@ OpResult<MScoreResponse> ZSetFamily::ZGetMembers(CmdArgList args, ConnectionCont
   return cntx->transaction->ScheduleSingleHopT(std::move(cb));
 }
 
+double ZSetFamily::ExtractUnitOrReply(std::string_view arg, ConnectionContext* cntx) {
```
Comment (romange) [inline src/server/zset_family.cc:2396]: it does not have to be ZSetFamily member, just add an auxillary function in anonymous namespace in this file.

File: src/server/zset_family.cc [hunk 2]
```diff
@@ -2471,6 +2486,41 @@ void ZSetFamily::GeoPos(CmdArgList args, ConnectionContext* cntx) {
...
+    return (*cntx)->SendError(result.status());
+  }
+
+  const MScoreResponse& arr = result.value();
+
+  if (arr.size() != 2) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+
+  double xyxy[4];
```
Comment (romange) [inline src/server/zset_family.cc:2514]: add a comment explaining that it's 2 pairs of scores holding two locations

File: src/server/zset_family.h [hunk 3]
```diff
@@ -93,10 +93,12 @@ class ZSetFamily {
   static bool ParseRangeByScoreParams(CmdArgList args, RangeParams* params);
   static void ZPopMinMax(CmdArgList args, bool reverse, ConnectionContext* cntx);
   static OpResult<MScoreResponse> ZGetMembers(CmdArgList args, ConnectionContext* cntx);
+  static double ExtractUnitOrReply(std::string_view arg, ConnectionContext* cntx);
```
Comment (romange) [inline src/server/zset_family.h:96]: no need

File: src/server/zset_family.cc [hunk 4]
```diff
@@ -2471,6 +2471,42 @@ void ZSetFamily::GeoPos(CmdArgList args, ConnectionContext* cntx) {
...
+  double to_meter = 1;
+  if (args.size() == 4) {
+    ToLower(&args[3]);
+    string_view unit = ArgS(args, 3);
+    to_meter = ExtractUnit(unit);
+    args.remove_suffix(1);
+    if (to_meter < 0) {
+      return (*cntx)->SendError("unsupported unit provided. please use M, KM, FT, MI");
+    }
+  } else if (args.size() < 3 || args.size() > 4) {
```
Comment (chakaz) [inline src/server/zset_family.cc:2484]: Maybe do `else if (args.size() != 3)`?

File: src/server/zset_family.h [hunk 5]
```diff
@@ -97,6 +97,26 @@ class ZSetFamily {
   static void GeoAdd(CmdArgList args, ConnectionContext* cntx);
   static void GeoHash(CmdArgList args, ConnectionContext* cntx);
   static void GeoPos(CmdArgList args, ConnectionContext* cntx);
+  static void GeoDist(CmdArgList args, ConnectionContext* cntx);
 };
 
 }  // namespace dfly
+
+namespace {
```
Comment (chakaz) [inline src/server/zset_family.h:105]: Anonymous namespaces shouldn't be in header files.
Please cut this code into the `.cc`, it's essentially implementation details.

File: src/server/zset_family.cc [hunk 6]
```diff
@@ -2471,6 +2471,42 @@ void ZSetFamily::GeoPos(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ZSetFamily::GeoDist(CmdArgList args, ConnectionContext* cntx) {
+  double to_meter = 1;
```
Comment (chakaz) [inline src/server/zset_family.cc:2475]: I'd rename this to `distance_multiplier` or something along those lines

File: src/server/zset_family.cc [hunk 7]
```diff
@@ -2471,6 +2471,42 @@ void ZSetFamily::GeoPos(CmdArgList args, ConnectionContext* cntx) {
...
+  if (args.size() == 4) {
+    ToLower(&args[3]);
```
Comment (chakaz) [inline src/server/zset_family.cc:2477]: Nit: Just a matter of convention, but we usually upper-case strings :shrug:

**Review by chakaz — COMMENTED** [review]
Thanks again for your contribution, @yoelsherwin! Some nits, but again LGTM!

**Comment by yoelsherwin** [comment]
@chakaz updated :)

**Review by chakaz — APPROVED** [review]
Thanks @yoelsherwin ! LGTM :)

**Pull request merged**